mobile/src/components/features/channels/channel-members/ViewChannelMembers.tsx

```diff
@@ -0,0 +1,162 @@
+import { IonAlert, IonButton, IonButtons, IonContent, IonHeader, IonItem, IonItemOption, IonItemOptions, IonItemSliding, IonLabel, IonList, IonModal, IonSearchbar, IonTitle, IonToolbar, ToastOptions, useIonToast } from "@ionic/react"
+import { useFrappeDeleteDoc, useFrappeGetCall } from "frappe-react-sdk"
+import { useContext, useMemo, useRef, useState } from "react"
+import { ChannelMembers, Member } from "./AddChannelMembers"
+import { SquareAvatar } from "@/components/common/UserAvatar"
+import { UserContext } from "@/utils/auth/UserProvider"
+import { ErrorBanner } from "@/components/layout"
+
+interface ViewChannelMembersProps {
+    presentingElement: HTMLElement | undefined,
+    isOpen: boolean,
+    onDismiss: VoidFunction,
+    channelID: string
+}
+
+export const ViewChannelMembers = ({ presentingElement, isOpen, onDismiss, channelID }: ViewChannelMembersProps) => {
+
+    const modal = useRef<HTMLIonModalElement>(null)
+    const { currentUser } = useContext(UserContext)
+
+    const { data, error, mutate } = useFrappeGetCall<{ message: ChannelMembers }>('raven.api.chat.get_channel_members', {
+        channel_id: channelID
+    }, undefined, {
+        revalidateOnFocus: false
+    })
+
+    const channelMembers = useMemo(() => {
+        if (data?.message) {
+            return Object.values(data.message)
+        } else {
+            return []
+        }
+    }, [data])
+
+    const isCurrentUserAdmin = useMemo(() => {
+        return channelMembers.find(member => member.name === currentUser)?.is_admin
+    }, [channelMembers, currentUser])
+
+    const [searchText, setSearchText] = useState('')
+    const [selectedMember, setSelectedMember] = useState<Member | undefined>(undefined)
+
+    const { data: member, error: errorFetchingChannelMember } = useFrappeGetCall<{ message: { name: string } }>('frappe.client.get_value', {
```

Comment:
Why is this API call required? This is also making an API call even if there's no selected member.

Reply:
added custom API